### DISCUSSION TITLE
How to connect via URL string to postgresql via local unix socket? percent-encoded absolute path are not supported

### DISCUSSION THREAD

**Discussion by thomas-riccardi** [sqlalchemy/sqlalchemy#10198] [Usage Questions] [answered]
postgresql connection URI support `host` as percent-encoded absolute paths to a unix socket: https://www.postgresql.org/docs/current/libpq-connect.html#id-1.7.3.8.3.6

But sqlalchemy does not seem to support that when connecting via a string.
```python
sqlalchemy.engine.make_url('postgresql://%2Ffoo/db').host
```
- actual output: ` '%2Ffoo'`
- expected output: `'/foo'`

It's related to the `@` escaping issue in the password part, which was fixed in 1.4.

https://github.com/sqlalchemy/sqlalchemy/blob/1377f33075a0e4230c88506d2bfd1e847a336da5/lib/sqlalchemy/engine/url.py#L885-L895

usage context:
- Celery ~requires a string, so passing a manually created `URL` object it not easy (sqlalchemy does support connection to postgresql via unix socket when passing via a `URL` object)
- proxies (gcp cloud-sql-proxy, pgbouncer) in sidecar containers in kubernetes

**Comment by CaselIT** [comment, marked as answer]
Hi,

A similar question was asked recently in https://github.com/sqlalchemy/sqlalchemy/discussions/8261

You probably want to use the host query param, so the form `postgresql:///dbname?host=/var/lib/postgresql`

**Reply by thomas-riccardi** [reply]
I confirm this works with the host in query string.

Furthermore: sqlalchemy (at least 1.4) uses the `host` from query string with higher priority than from netloc, if both are present.

This could be documented, but at least it's here now.

Thank you.

**Reply by CaselIT** [reply]
it's documented here https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#unix-domain-connections

@zzzeek maybe it could make sense to move these connection string docs out of psycopg2? They seem to be quite uniform across the other dielects

**Reply by zzzeek** [reply]
does it work for pg8000?   We might want to just continue adding sections to psycopg and asyncpg that point to psycopg2.   pg8000 is likely an outlier here so these aren't generic docs (multihost, SSL, unix hosts, all of these things)

**Reply by CaselIT** [reply]
Doing the opposite? 'these following options are not known yo work on pg8000'?

**Reply by zzzeek** [reply]
not really sure, im proposing we check before making a generic section.   I think mutlihost/ssl/unix host should be subsections in a basic section on "connecting"

**Reply by CaselIT** [reply]
I'll try proposing something
